modules/@apostrophecms/attachment/index.js

```diff
@@ -1037,6 +1042,66 @@ module.exports = {
           });
         }
         next();
+      },
+      // This migration is needed because formerly,
+      // docs that only referenced this attachment via
+      // a join were counted as "owning" it, which is
+      // incorrect and leads to failure to make it
+      // unavailable at the proper time. The name was
+      // changed to ensure this migration would run
+      // again after that bug was discovered and fixed.
+      addDocReferencesContainedMigration() {
+        self.apos.migration.add(self.__meta.name + '.docReferencesContained', self.recomputeAllDocReferences);
+      },
+      // Recompute the `docIds` and `archivedDocIds` arrays
+      // from scratch. Should only be needed by the
+      // one-time migration that fixes these for older
+      // databases, but can be run at any time via the
+      // `apostrophe-attachments:recompute-doc-references`
+      // task, just in case the need arises or your site
+      // was affected by the very brief availability of 2.77.0
```

Comment:
Is this `2.77.0` version number right? This is for 3.x

Reply:
It was accurate but a reference to a very old incident not really relevant in 3.x source comments, so removed.